encryptonize.go

```diff
@@ -430,3 +433,35 @@ func (e *Encryptonize) GetGroupData(authorizer *SealedUser, group *SealedGroup)
 
 	return plainGroup.Data, nil
 }
+
+////////////////////////////////////////////////////////
+//                       Index                        //
+////////////////////////////////////////////////////////
+
+// NewIndex creates an instance of index which is a building block of searchable encryption.
+// An index contains a mapping from labels to sealedIDs. A label is based on a keyword, and
+// a sealedID is based on the occurrence of that keyword in a specific ID (e.g. a document).
+// Hence, index makes it possible to manage which keywords are contained in which IDs.
+func (e *Encryptonize) NewIndex() index {
+	return newIndex()
+}
+
+// Add adds the keyword/id pair to Index i.
+func (e *Encryptonize) Add(keyword, id string, i *index) error {
+	if err := i.add(e.IndexKey, keyword, id); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Given a keyword, Search returns all decrypted ids (contained in Index i)
```

Comment:
Docstrings should start with the name of the thing they document, e.g. "Search finds all IDs that contain the given keyword..."